### PR TITLE
Type aliasing

### PR DESCRIPTION
**Pull request by dmorilha**
@sc0ttbeardsley this PR implements type aliasing, yaml `maps` can now alias an extracted type just by simply annotating its master entry. The feature has no current implementation to the Java back-end and it is simply ignored.

### CODE REVIEW

File: conf/04.master.yaml [hunk 1]
```diff
@@ -5,14 +5,14 @@
 ---
 - settings: master
 
-  provider:
+  provider: !Provider
```
Comment (sc0ttbeardsley) [inline conf/04.master.yaml:8]: is this format required now?

Comment (sc0ttbeardsley) [inline conf/04.master.yaml:8]: nevermind... explained offline. Aliasing is just to help out the devs

**Comment by sc0ttbeardsley** [comment]
:+1:

File: conf/04.master.yaml (hunk 1 above)
Comment (dmorilha) [inline conf/04.master.yaml:8]: it is all optional

**Comment by sc0ttbeardsley** [comment]
@dmorilha for Java what about extending the generated class? It would be a bit slower but should allow devs to use the alias as a class name.

**Pull request merged**